File: my-folder/0242-valid-anagram/solution.rs
use std::collections::HashMap;
impl Solution {
    pub fn is_anagram(s: String, t: String) -> bool {
        let iter = s.chars().zip(t.chars());
        let result = iter
            .fold(HashMap::new(), |mut map, (x, y)| {
                map.entry(x).and_modify(|e| *e += 1).or_insert(1);
                map.entry(y).and_modify(|e| *e -= 1).or_insert(-1);
                map
            });
        s.len() == t.len() && !result.values().any(|&x| x != 0)
    }
}

File: my-folder/0242-valid-anagram/solution.c
bool isAnagram(char* s, char* t) {
    size_t hashmap[256] = {0};

    while (*s && *t) {
        hashmap[*s]++;
        hashmap[*t]--;
        s++; t++;
    }

    if (*s != *t)
        return false;

    for (size_t i = 0; i < 256; i++) {
        if (hashmap[i] != 0)
            return false;
    }

    return true;
}
